### PR TITLE
Add WindbarbVisual

### PR DESCRIPTION
**Pull request by kmuehlbauer**
This is a follow up PR from #1563. 

This adds a special WindbarbVisual as well as an example. The Visual might need some tweaking here and there. Would be nice if someone can test and review. There is also shader code which can be moved to the gsls subfolder (see also #1566).

Ping @sjoro and @rougier.

@larsoner @djhoese How and where should I add tests for this? Is the example tested anyway?

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@sjoro @djhoese This is now ready for review.

**Comment by sjoro** [comment]
i'll try and test this hopefully next week.

**Comment by djhoese** [comment]
@sjoro Did you ever get a chance to test this?

**Comment by kmuehlbauer** [comment]
@djhoese Where should I add tests here? Via image comparison? Thoughts?

**Comment by djhoese** [comment]
I'd be ok with image comparison. If you can make the image/buffer small enough maybe do raw array comparison without having to add images to vispy-data?

**Comment by kmuehlbauer** [comment]
@djhoese  Yes, if I only consider one barb this might work. Let me try this.

**Comment by sjoro** [comment]
sorry guys, i haven't tested this yet. hope to have some time soon.

**Comment by kmuehlbauer** [comment]
@djhoese Raw Image comparison isn't feasible, but I think utilizing `numpy.histogram` should do the trick. I rebased on current master.

**Comment by kmuehlbauer** [comment]
So the histogram differs from my local histogram, but in a consistent way. I just might use those values then.

**Comment by kmuehlbauer** [comment]
Some OT:
@djhoese @larsoner One thing which I'm thinking about is using something like `phash` (eg. https://github.com/JohannesBuchner/imagehash) to test creation of images. That way it would be possible to get rid of the `test-data` repo while still test for correctness of created images via hashes. If there is interest, I would check further and try to implement that alongside the current implementation.

**Comment by larsoner** [comment]
Not sure it's worth it. We wouldn't have to download things on CIs but is this a problem? Is there some other problem it would fix?

A few concerns that occur to me off the top of my head:

- The image hashing would need to be sufficiently flexible in the ways that our current checkers are
- When there are failures hopefully it should be at least as informative as to why
- If we switch, we are stuck with whatever ways the image hasher is flexible with no ways to add new types of flexibility
- When there are failures, currently you can get a nice side-by-side of the desired and actual plus diff, this would not be possible without having `test-data`

**Comment by kmuehlbauer** [comment]
@larsoner Thanks, this all makes sense. 

Then I'll need to add the image for this visual here to the vispy-test repo. I can't make the image(s) small enough to create raw array tests and the `histogram` approach is obviously not robust enough.

**Comment by djhoese** [comment]
I agree with @larsoner. As much as it feels wrong to me to have a separate repository for storing test/compare images, having `--vispy-audit` functionality (I think that's the flag) for comparing images when tests fail is pretty nice.

File: vispy/visuals/__init__.py [hunk 1]
```diff
@@ -44,3 +45,4 @@
 from .border import _BorderVisual  # noqa
 from .colorbar import ColorBarVisual  # noqa
 from .graphs import GraphVisual  # noqa
+from .windbarb import WindbarbVisual  # noqa
```
Comment (djhoese) [inline vispy/visuals/__init__.py]: Why is the visual imported twice?

Comment (kmuehlbauer) [reply vispy/visuals/__init__.py]: Not sure, two are better than one? I'll clean-up.

**Comment by kmuehlbauer** [comment]
windbarb

**Comment by kmuehlbauer** [comment]
@djhoese Is there something wrong with travis? I rebased on latest master, but travis seems not to start.

**Comment by larsoner** [comment]
I thought that closing and reopening might fix it but it didn't. Let's just make sure to check the Travis status before merge

**Comment by kmuehlbauer** [comment]
@larsoner It worked! Let's see, if the image test works correctly, hopefully.

**Comment by kmuehlbauer** [comment]
@larsoner Do you have any idea, why the image test might fail. It worked locally...

**Comment by larsoner** [comment]
Travis:
```E                   Exception: Test standard /Users/travis/Library/Application Support/vispy/test_data/visuals/windbarb.png does not exist.
```

Did you follow all steps in https://github.com/vispy/vispy/blob/master/vispy/testing/image_tester.py#L4 ?

**Comment by kmuehlbauer** [comment]
Did not increment the tag...

**Comment by larsoner** [comment]
Do you have permissions to do it, or do you need me to?

**Comment by kmuehlbauer** [comment]
AFAIKT I would also need to increase that number in https://github.com/vispy/vispy/blob/6894b3fa81c043cab545c4e5350ce99df86bf89c/vispy/testing/image_tester.py#L367

I'll try to add the tag and do the other change on top of this PR.

**Comment by kmuehlbauer** [comment]
Created the tag in test-data and changed it accordingly in `get_test_data_repo`. :crossed_fingers:

**Comment by kmuehlbauer** [comment]
Nice, now the green travis build is linked correctly in the checks above. :green_heart:

**Comment by kmuehlbauer** [comment]
@larsoner Thanks for the help with the test-data. @djhoese This is ready for final review/merge.

**Comment by djhoese** [comment]
Looks good to me. Great work!

**Comment by sjoro** [comment]
just tried your windbarb example @kmuehlbauer, looks great! thank you for your great work!

**Comment by kmuehlbauer** [comment]
@sjoro Glad you find it useful!

**Pull request merged**